apps/block_scout_web/lib/block_scout_web/channels/stakes_channel.ex

```diff
@@ -658,6 +691,28 @@ defmodule BlockScoutWeb.StakesChannel do
     end
   end
 
+  defp calc_apy(pool, staker, pool_staking_address_downcased, pool_reward, average_block_time, staking_epoch_duration) do
+    staker_address = String.downcase(to_string(staker.address_hash))
+
+    if staker_address == pool_staking_address_downcased do
+      ContractState.calc_apy(
+        pool.snapshotted_validator_reward_ratio,
+        pool_reward,
+        pool.snapshotted_self_staked_amount,
+        average_block_time,
+        staking_epoch_duration
+      )
+    else
+      ContractState.calc_apy(
+        staker.snapshotted_reward_ratio,
+        pool_reward,
+        staker.snapshotted_stake_amount,
+        average_block_time,
+        staking_epoch_duration
+      )
+    end
+  end
```

Comment:
Refactor suggestion (just for readability):

```suggestion
    if staker_address == pool_staking_address_downcased do
      reward_ratio = pool.snapshotted_validator_reward_ratio
      staked_amount = pool.snapshotted_self_staked_amount
    else
      reward_ratio = staker.snapshotted_reward_ratio
      staked_amount = staker.snapshotted_stake_amount
    end
    ContractState.calc_apy(
        reward_ratio,
        pool_reward,
        staked_amount,
        average_block_time,
        staking_epoch_duration
      )
  end
```

Reply:
Done in https://github.com/poanetwork/blockscout/pull/3597/commits/b408a6821a0ec71590a2c6fabdd6e1aae2e1e896